### PR TITLE
Added swiftpm and other libs to Android toolchain

### PR DESCRIPTION
**Pull request by amraboelela**
These changes will add swift package management tools to the android toolchain, among other libraries and tools

### CODE REVIEW

**Comment by amraboelela** [comment]
@milseman Please check this :)

**Comment by milseman** [comment]
@Rostepher, could you review this or let me know who is a more appropriate reviewer for this?

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by amraboelela** [comment]
Question. When you say @ swift-ci please smoke test, are there humans behind this or is it automatic thing? :)

**Comment by milseman** [comment]
Automatically handled by a bot. There are sometimes issues, of course, which are inherent to the problem of CI.

**Comment by Rostepher** [comment]
If you don't see a build start almost immediately then the CI bot probably didn't get the message.

**Comment by amraboelela** [comment]
Does that mean I can also initiate the test by saying @ swift-ci please smoke test ?

**Comment by milseman** [comment]
@amraboelela  It might be limited to those with commit access, to help manage potential load on the bots.

**Comment by amraboelela** [comment]
@swift-ci please smoke test

**Comment by amraboelela** [comment]
Just testing :) so I guess it didn't work from me.

**Comment by amraboelela** [comment]
Oh, the tests actually got started after a while, so maybe anybody can start them.

**Comment by amraboelela** [comment]
@swift-ci please smoke test

**Comment by amraboelela** [comment]
It got stuck at Waiting for status to be reported

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
Not that I know what I'm taking about, but is it reasonable to constrain the script to only building for armv7e?  Shouldn't this be something that's figured out from a target triple instead of hard-coded into a script?

**Comment by milseman** [comment]
@Rostepher could you review this?

**Comment by compnerd** [comment]
@CodaFi android only supports the A class of CPUs AFAIK.  The NDK only provides the v7a bits, so I think that bit is reasonable.

**Comment by Rostepher** [comment]
@milseman I'll give this a look over in the morning.

**Review by Rostepher — COMMENTED** [review]
I don't see any issues with this, except for what @CodaFi pointed out with hardcoding the ARM version. I've not had any experience with building for Android yet so take my input with a grain of salt. @gottesmm Do you have any experience with Android builds?

**Comment by gottesmm** [comment]
No. I think @modocache is the best AFAIK.

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
@amraboelela note that the swift-ci bot is picky about who it listens to - committers have access to trigger builds but not others. When it says 'waiting for status' it's really waiting for someone else to kick it off on your behalf.

**Comment by milseman** [comment]
@compnerd or @modocache what do you think of this change?

File: utils/android/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+# Build Android Toolchain
```
Comment (modocache) [inline utils/android/README.md]: Just a suggestion: I think expanding on what "Android toolchain" means here would help people who have some interest, but don't quite understand what a "toolchain" is, to use the scripts and tools in here. Ideally, I think this README should answer the question: "why would I want to build an Android toolchain?"

**Review by modocache — APPROVED** [review]
This is great, thanks!

I believe "armeabi-v7a" is hardcoded in a few spots in the build scripts and testing utilities for Android, so it doesn't bother me that it also is here. Maybe someone could create a task to clean that up and take care of it in a forthcoming pull request?

File: utils/android/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+This toolchain will generate the .so and .swiftmodule files of swift standard library and Foundation framework for Android environment, armv7 architecture. Those files are needed when building any swift library to be included in an applciation for Android.
```
Comment (xwu) [inline utils/android/README.md]: Please capitalize "Swift" and correct the typo in the word "application".
Insert "the" before "Swift standard library" and before "Android environment".

**Comment by milseman** [comment]
@amraboelela, do you want to do those fixes in this PR or a followup? I know it's been open for quite a while, and I have no strong opinions so long as @modocache and @xwu are fine with either.

**Comment by amraboelela** [comment]
@milseman I already did the changes that @modocache and @xwu asked me to do.

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by alblue** [comment]
@swift-ci please smoke test

**Comment by milseman** [comment]
@swift-ci please smoke test Linux Platform

File: utils/android/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+# Build Android Toolchain
+
+This toolchain will generate the .so and .swiftmodule files of the Swift standard library and Foundation framework for the Android environment, armv7 architecture. Those files are needed when building any swift library to be included in an application for Android.
```
Comment (xwu) [inline utils/android/README.md]: Please capitalize all instances of "Swift" :)

**Comment by milseman** [comment]
@swift-ci please smoke test

**Comment by amraboelela** [comment]
Did I ruin the smoke test by doing merge from master?

**Comment by Rostepher** [comment]
Any new commits will update the status of the PR and require a new smoke-test run. Same goes for force pushing a `git rebase` or a new merge commit.

**Comment by Rostepher** [comment]
@swift-ci please smoke test

**Comment by amraboelela** [comment]
Did another merge from master because the Linux smoke test failed.

**Comment by modocache** [comment]
@swift-ci please smoke test

**Pull request merged**